zebedee-cms/src/test/java/com/github/onsdigital/zebedee/api/ContentTest.java

```diff
@@ -200,8 +217,9 @@ public void writeVersionFileV2() throws Exception {
             throw e;
         }
 
-        Assert.assertEquals(200, response.getStatus());
-        JSONAssert.assertEquals(EXPECTED_CONTENT_V2, response.getContentAsString(), false);
+        // Then I should receive a structured V1 content back with uri download link
```

Comment:
I still think we don't need 2 versions of this test but I suppose that it will be addressed in a separate PR (if addressed at all), for now this one should refer to V2
```suggestion
        // Then I should receive a structured V2 content back with uri download link
```

Reply:
There are two separate behaviours of the system because it deserialises differently for each type of content, so there need to be two separate tests. 
I corrected the comment, thanks.

Reply:
I think I am missing something here as I can't see any difference in these tests. I'd appreciate if you or anyone in the team could explain better? Why would not [something like this](https://github.com/ONSdigital/zebedee/compare/wip/single-test) be acceptable?

Reply:
That does represent what should happen.

The download will either have `file` or (`uri` and `version`) that suggested test represents an illegal state.

Reply:
But the two test represent that way that this part of the JSON should be use. V1 format (`file`) is the current way of working when the files are store and server in Zebedee. V2 format (`uri` and `version`) mean that the file is store and served by by the static files system (dp-upload, dp-download, dp-files).

The test should that both versions can work in parallel. The two test are showing two different behaviours of the same endpoint. 

If the data where in the illegal state (having all three fields set at the same time or having `version` and `files` set but not `url`) this would have the potential of breaking downstream systems of zebedee.

Reply:
Thanks, I understand that but it is not an illegal state as there are no rules checking it and hence the test with the 3 fields passes. The two tests do not show any different behaviour but only one: it can process the request by marshalling/unmarshalling the `DownloadSection`. I do not think there is any different logic in the endpoint when it receives a `version` and `files` than when it receives an `url`. It is dumb as far as I can see and we are only making sure it can accept the new fields. Having 2 tests doing the same thing (and not even sharing the code) is only creating technical debt and unnecessary maintenance.
But that is not really part of this pull request and should have been discussed and addressed earlier (see https://github.com/ONSdigital/zebedee/pull/621) instead of ignoring the review comments. On that basis, I don't think it should stop this branch from being merged 👍